zebra-state/src/service/finalized_state/zebra_db/block.rs

```diff
@@ -638,15 +638,14 @@ impl DiskWriteBatch {
                 &out_loc_by_outpoint,
                 address_balances,
             )?;
-
-            // Commit UTXOs and value pools
-            self.prepare_chain_value_pools_batch(
-                zebra_db,
-                finalized,
-                spent_utxos_by_outpoint,
-                value_pool,
-            )?;
         }
+        // Commit UTXOs and value pools
+        self.prepare_chain_value_pools_batch(
+            zebra_db,
+            finalized,
+            spent_utxos_by_outpoint,
+            value_pool,
+        )?;
```

Comment:
This is a bit of subtle change. Per the comment above (expand to see), we should ignore the genesis block. But for the value pools it's harmless, because the genesis block adds 0 balance and it does not hurt storing it for block 0; in fact, doing so makes some stuff easier to handle later on (such as tests, where we don't need to special-case block 0). I might have missed something though, so please double check my reasoning, but it seems OK.